src/parquet/column/properties.h

```diff
@@ -0,0 +1,153 @@
+
+class ReaderProperties {
+ public:
+  ReaderProperties(bool use_buffered_stream = false, int64_t buffer_size = 0,
+      MemoryAllocator* allocator = default_allocator()) {
+    use_buffered_stream_ = use_buffered_stream;
+    buffer_size_ = buffer_size;
+    allocator_ = allocator;
+  }
+
+  static inline std::string type_printer(Type::type parquet_type, const char* val,
+      int length) {
+    std::stringstream result;
+    switch (parquet_type) {
+      case Type::BOOLEAN:
+        result << reinterpret_cast<const bool*>(val)[0];
+        break;
+      case Type::INT32:
+        result << reinterpret_cast<const int32_t*>(val)[0];
+        break;
+      case Type::INT64:
+        result << reinterpret_cast<const int64_t*>(val)[0];
+        break;
+      case Type::DOUBLE:
+        result << reinterpret_cast<const double*>(val)[0];
+        break;
+      case Type::FLOAT:
+        result << reinterpret_cast<const float*>(val)[0];
+        break;
+      case Type::INT96: {
+        for (int i = 0; i < 3; i++) {
+          result << reinterpret_cast<const int32_t*>(val)[i] << " ";
+        }
+        break;
+      }
+      case Type::BYTE_ARRAY: {
+        const ByteArray* a = reinterpret_cast<const ByteArray*>(val);
+        for (int i = 0; i < static_cast<int>(a->len); i++) {
+          result << a[0].ptr[i]  << " ";
+        }
+        break;
+      }
+      case Type::FIXED_LEN_BYTE_ARRAY: {
+        const FLBA* a = reinterpret_cast<const FLBA*>(val);
+        for (int i = 0; i < length; i++) {
+          result << a[0].ptr[i]  << " ";
+        }
+        break;
+      }
+      default:
+        break;
+    }
+    return result.str();
+  }
+
+  MemoryAllocator* get_allocator() {
+    return allocator_;
+  }
+
+  std::unique_ptr<InputStream> GetInputStream() {
+    std::unique_ptr<InputStream> stream;
+    if (use_buffered_stream_) {
+      stream.reset(new BufferedInputStream(allocator_, buffer_size_));
+    } else {
+      stream.reset(new InMemoryInputStream());
+    }
+    return stream;
+  }
+
+  bool use_buffered_stream() {
+    return use_buffered_stream_;
+  }
+
+  int64_t get_buffer_size() {
+    return buffer_size_;
+  }
+
+ private:
+  MemoryAllocator* allocator_;
+  bool use_buffered_stream_;
+  int64_t buffer_size_;
+};
+
+class WriterProperties {
+ public:
+  WriterProperties(int pagesize = DEFAULT_PAGE_SIZE,
+      int dictionary_pagesize = DEFAULT_DICTIONARY_PAGE_SIZE,
+      bool enable_dictionary = DEFAULT_IS_DICTIONARY_ENABLED,
+      MemoryAllocator* allocator = default_allocator()) {
+    pagesize_ = pagesize;
+    dictionary_pagesize_ = dictionary_pagesize;
+    enable_dictionary_ = enable_dictionary;
+    allocator_ = allocator;
+  }
+
+  int get_dictionary_pagesize_threshold() {
+    return dictionary_pagesize_;
+  }
+
+  int get_pagesize_threshold() {
+    return pagesize_;
+  }
+
+  bool is_enable_dictionary() {
+    return enable_dictionary_;
+  }
+
+  MemoryAllocator* get_allocator() {
```

Comment:
Also here. We might differ then interface wise from Java but are more consitent with the existing ones.